algorithms task (4)/student/include/impl.h

```diff
@@ -1,14 +1,38 @@
 #pragma once
+
 #include <algorithm>
+#include <iterator>
+#include <utility>
+#include <vector>
 
 /**
  * @todo Implement sliding algorithm (left/right shift of subsequence to a position)
  *
  * @param begin - begin of random-access range
- * @param last - end of random-access range
- * @param pos - new begin of slided range
+ * @param end - end of random-access range
+ * @param newEnd - new end of slided range
  *
  * @return new slided range position
  */
 template <typename RandomAccessIterator>
-std::pair<RandomAccessIterator, RandomAccessIterator> slide(RandomAccessIterator begin, RandomAccessIterator last, RandomAccessIterator pos);
+std::pair<RandomAccessIterator, RandomAccessIterator> slide(RandomAccessIterator begin, RandomAccessIterator end, RandomAccessIterator newEnd)
+{
+    const auto rangeSize = std::distance(begin, end);
+    const auto newBegin = std::prev(newEnd, rangeSize);
+
+    std::vector<typename std::iterator_traits<RandomAccessIterator>::value_type> tmp;
```

Comment:
1. algorithms usually use get_temporary_buffer (it was designed for it)
2. consider renaming

Reply:
1. Got it!
2. Nothing to rename :)